pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/player/SimplePlayerViewModel.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2022. SRG SSR. All rights reserved.
+ * License information is available from the LICENSE file.
+ */
+package ch.srgssr.pillarbox.demo.ui.player
+
+import android.app.Application
+import android.util.Log
+import androidx.lifecycle.AndroidViewModel
+import androidx.media3.common.MediaItem
+import androidx.media3.common.MediaMetadata
+import androidx.media3.common.PlaybackException
+import androidx.media3.common.PlaybackParameters
+import androidx.media3.common.Player
+import androidx.media3.common.Timeline
+import ch.srgssr.pillarbox.demo.data.DemoItemDataSource
+import ch.srgssr.pillarbox.demo.data.DemoMediaItemSource
+import ch.srgssr.pillarbox.player.PillarboxPlayer
+
+/**
+ * Simple player view model than handle a PillarboxPlayer [player]
+ */
+class SimplePlayerViewModel(application: Application) : AndroidViewModel(application), Player.Listener {
+    /**
+     * Player as PillarboxPlayer
+     */
+    val player = PillarboxPlayer(application, DemoMediaItemSource(DemoItemDataSource(application)))
+
+    init {
+        player.addListener(this)
+    }
+
+    /**
+     * Add to [player] all [ids] to the MediaItem list.
+     * Will prepare and play the content.
+     *
+     * @param ids mediaIdentifier to play
+     */
+    fun playItemIds(ids: Array<String>) {
+        player.addMediaItems(ids.map { fromMediaId(it) })
+        player.prepare()
+        player.play()
+    }
+
+    override fun onCleared() {
+        super.onCleared()
+        player.release()
+        player.removeListener(this)
+    }
+
+    /**
+     * Resume playback of [player]
+     */
+    fun resumePlayback() {
+        player.play()
+    }
+
+    /**
+     * Pause playback of [player]
+     */
+    fun pausePlayback() {
+        player.pause()
+    }
+
+    override fun onMediaMetadataChanged(mediaMetadata: MediaMetadata) {
+        Log.d(TAG, "onMediaMetadataChanged title = ${mediaMetadata.title}")
+    }
+
+    override fun onPlaylistMetadataChanged(mediaMetadata: MediaMetadata) {
+        Log.d(TAG, "onPlaylistMetadataChanged title = ${mediaMetadata.title}")
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/SRGSSR/pillarbox-android/blob/9e672e88664d24e42b2fbb65a69ff4507dff46b6/pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/player/SimplePlayerViewModel.kt#L70"></span>The log call Log.d(…) should be conditional: surround with <code>if (Log.isLoggable(...))</code> or <code>if (BuildConfig.DEBUG) { ... }</code></td>
    </tr>
  </tbody>
</table>